### PR TITLE
Add call to Starport Lambda to clear `/tmp`

### PR DESCRIPTION
**Pull request by jkbudd**

### CODE REVIEW

**Comment by jkbudd** [comment]
@realstraw Ready for review.

File: starport-lambda/src/main/scala/com/krux/starport/lambda/SubmitHandler.scala [hunk 1]
```diff
@@ -33,31 +32,42 @@ class SubmitHandler extends RequestHandler[SubmitRequest, SubmitResponse] with L
...
-      case caughtExit: LambdaExitException => {
-        status = caughtExit.status
-        logger.error("exit:", caughtExit)
-        logger.error(scanTmpFiles())
-      }
-      case unhandled: Throwable =>  {
+    val args: Array[String] = input.getArgs
+
+    args.head match {
+      case "deleteTmpDir" => {
```
Comment (realstraw) [inline starport-lambda/src/main/scala/com/krux/starport/lambda/SubmitHandler.scala:38]: this one should be in the `try` part below

File: starport-lambda/src/main/scala/com/krux/starport/lambda/SubmitHandler.scala [hunk 2]
```diff
@@ -69,5 +79,12 @@ class SubmitHandler extends RequestHandler[SubmitRequest, SubmitResponse] with L
       .map(f => s"${f.getAbsolutePath}: ${f.length()}")
       .mkString("\n")
   }
-}
 
+  /**
+   * @return delete the /tmp directory
+   */
+  private def deleteTmpDir(): String = {
+    new File("/tmp").delete
```
Comment (realstraw) [inline starport-lambda/src/main/scala/com/krux/starport/lambda/SubmitHandler.scala:87]: I thought we have a prefix? Might not be a good idea to delete every thing in `/tmp`

File: starport-lambda/src/main/scala/com/krux/starport/lambda/SubmitHandler.scala (hunk 1 above)
Comment (jkbudd) [reply starport-lambda/src/main/scala/com/krux/starport/lambda/SubmitHandler.scala]: Ah yes, sorry.

File: starport-lambda/src/main/scala/com/krux/starport/lambda/SubmitHandler.scala (hunk 2 above)
Comment (jkbudd) [reply starport-lambda/src/main/scala/com/krux/starport/lambda/SubmitHandler.scala:86]: There's no prefix in the `scanTmpFiles` method - I assumed we wanted the same directory.

Comment (realstraw) [reply starport-lambda/src/main/scala/com/krux/starport/lambda/SubmitHandler.scala:86]: let's create a subfolder in `/tmp`

Comment (jkbudd) [reply starport-lambda/src/main/scala/com/krux/starport/lambda/SubmitHandler.scala]: OK this is done now.

**Comment by jkbudd** [comment]
@realstraw This is ready for another look.

File: starport-core/src/main/scala/com/krux/starport/util/S3FileHandler.scala [hunk 3]
```diff
@@ -14,7 +14,7 @@ object S3FileHandler extends Logging {
 
   def getTempDirectory(baseDir: Option[String]): File = {
     if (Lambda.isLambda()) {
-      new File("/tmp")
+      new File("/tmp/.starport")
```
[the remark this answers is not in the document]
Comment (jkbudd) [reply starport-core/src/main/scala/com/krux/starport/util/S3FileHandler.scala]: OK done: temp directory location can now be specified as part of the config (with a default location) and can be accessed from `StarportSettings`.

File: starport-lambda/src/main/scala/com/krux/starport/lambda/SubmitHandler.scala [hunk 4]
```diff
@@ -33,41 +32,59 @@ class SubmitHandler extends RequestHandler[SubmitRequest, SubmitResponse] with L
...
 
-    SubmitResponse(outString, errString, status, input)
   }
 
   /**
    * @return /tmp file listing with size for troubleshooting purposes
    */
   private def scanTmpFiles(): String = {
-    new File("/tmp")
+    new File("/tmp/.starport")
```
Comment (realstraw) [inline starport-lambda/src/main/scala/com/krux/starport/lambda/SubmitHandler.scala:76]: I mean it's referenced here too. We already accept different tmp directories. Define a constant val, and reference it in both places

Comment (jkbudd) [reply starport-lambda/src/main/scala/com/krux/starport/lambda/SubmitHandler.scala:76]: Ah gotcha. I'll fix that now.

File: starport-core/src/main/scala/com/krux/starport/util/S3FileHandler.scala [hunk 5]
```diff
@@ -5,16 +5,15 @@ import java.net.URI
 
 import com.amazonaws.services.s3.AmazonS3ClientBuilder
 import com.krux.starport.Logging
-import com.krux.starport.config.StarportSettings
 
 object S3FileHandler extends Logging {
 
   val bufferSize = 1024
-  lazy val config = StarportSettings()
+  final val tmpDirectory = "/tmp/starport"
```
Comment (realstraw) [inline starport-core/src/main/scala/com/krux/starport/util/S3FileHandler.scala:12]: let's start with capital letter `TmpDirectory` for constants

File: starport-core/src/main/scala/com/krux/starport/util/S3FileHandler.scala [hunk 6]
```diff
@@ -4,17 +4,16 @@ import java.io.{BufferedOutputStream, File, FileOutputStream}
...
-
 object S3FileHandler extends Logging {
 
   val bufferSize = 1024
+  final val TmpDirectory = "/tmp/starport"
 
   def getTempDirectory(baseDir: Option[String]): File = {
     if (Lambda.isLambda()) {
-      new File("/tmp")
+      new File(tmpDirectory)
```
Comment (realstraw) [inline starport-core/src/main/scala/com/krux/starport/util/S3FileHandler.scala:16]: `TmpDirectory`

File: starport-lambda/src/main/scala/com/krux/starport/lambda/SubmitHandler.scala [hunk 7]
```diff
@@ -27,47 +25,65 @@ class SubmitHandler extends RequestHandler[SubmitRequest, SubmitResponse] with L
...
 
-    SubmitResponse(outString, errString, status, input)
   }
 
   /**
    * @return /tmp file listing with size for troubleshooting purposes
    */
   private def scanTmpFiles(): String = {
-    new File("/tmp")
+    new File(tmpDirectory)
```
Comment (realstraw) [inline starport-lambda/src/main/scala/com/krux/starport/lambda/SubmitHandler.scala:76]: `S3FileHandler.TmpDirectory`

File: starport-lambda/src/main/scala/com/krux/starport/lambda/SubmitHandler.scala [hunk 8]
```diff
@@ -27,47 +25,65 @@ class SubmitHandler extends RequestHandler[SubmitRequest, SubmitResponse] with L
   val lambdaErr = System.err
   System.setErr(errPrintStream)
   System.setOut(outPrintStream)
+  final val TmpDirectory = "/tmp/starport"
```
Comment (realstraw) [inline starport-lambda/src/main/scala/com/krux/starport/lambda/SubmitHandler.scala:28]: no need to have this one, can reference `S3FileHandler.TmpDirectory `

**Comment by realstraw** [comment]
LGMT, can you update the version to 7.4.0?

**Comment by jkbudd** [comment]
Can do!

**Review by realstraw — APPROVED** [review]

**Comment by realstraw** [comment]
@jkbudd can you resolve conflicts too?

**Comment by jkbudd** [comment]
@realstraw Done diggity done

**Review by realstraw — APPROVED** [review]

**Pull request merged**